code/game/machinery/_machinery.dm

```diff
@@ -506,3 +506,11 @@ Class Procs:
 	. = ..()
 	if (AM == occupant)
 		occupant = null
+
+/obj/machinery/proc/adjust_item_drop_location(atom/movable/AM)	// Adjust item drop location to a 3x3 grid inside the tile, returns slot id from 0 to 8
+	var/md5 = md5(AM.name)										// Oh, and it's deterministic too. A specific item will always drop from the same slot.
+	for (var/i in 1 to 32)
+		. += hex2num(copytext(md5,i,i+1))
```

Comment:
Add a compile error if DM_VERSION >= 512 so we can use string iteration eventually